### ISSUE TITLE
Exception when accept() returns False

### ISSUE THREAD

**Issue by pedrudehuere** [zatosource/zato#437]
I've got an error when I override the accept() method of a Service and it returns False,
the server log gives:
```
2015-04-23 15:52:59,978 - ERROR - 14492:Dummy-1069 - zato.server.connection.http_soap.channel:22 - Caught an exception, cid:[K046E8FASHG942ZDKTXP98H5532E], status_code:[500], _format_exc:[Traceback (most recent call last):
  File "/home/altux/zato-2.0.3/code/zato-server/src/zato/server/connection/http_soap/channel.py", line 201, in dispatch
    self.add_response_headers(wsgi_environ, response)
  File "/home/altux/zato-2.0.3/code/zato-server/src/zato/server/connection/http_soap/channel.py", line 271, in add_response_headers
    wsgi_environ['zato.http.response.headers']['Content-Type'] = response.content_type
AttributeError: 'NoneType' object has no attribute 'content_type'
```

**Comment by dsuch** [comment]
Aha, I can see how it can happen.

I believe I didn't realize that the ```def accept(self)``` feature would be used with services invoked synchronously.

I mean, what should the output really be if the request is to be dropped?

What to return in an HTTP body and headers?

**Comment by pedrudehuere** [comment]
Eheh, I was asking myself the same question. Could you tell me about a use case for this feature (or point me to some doc mentioning it?)?

Thanks.

**Comment by dsuch** [comment]
Initially, it was designed for use in a heavy async environment where HTTP clients would get quick "OK, got your request, here is CID .." messages and all of the processing was performed as a set of callbacks and parallel async paths with an eventual response being delivered to the initial client by a different service than the originally invoked one.

At times it was handy to drop certain messages - but with sync invocations, hm..

Well, it's a sync so there will be some response in any case. It's better to return something useful instead of raising a 500 exception. For 2.0.4 we can make it return an empty string + HTTP 400. That, I believe, makes most sense. In 2.1 we can make it configurable.

**Comment by pedrudehuere** [comment]
makes totally sense, thanks!

**Comment by dsuch** [comment]
Thanks @pedrudehuere - this was done in 5ee721a0d239f0eaefdd6bf92193050af4b693e0. In master + backported to support/2.0, will be released in 2.0.4.
